Replicator/IncomingRev.cc

```diff
@@ -452,12 +452,27 @@ namespace litecore::repl {
         _bodySize       = 0;
     }
 
-    Worker::ActivityLevel IncomingRev::computeActivityLevel() const {
-        if ( Worker::computeActivityLevel() == kC4Busy || _pendingCallbacks > 0 || (_blob != _pendingBlobs.end()) ) {
-            return kC4Busy;
+    Worker::ActivityLevel IncomingRev::computeActivityLevel(std::string* reason) const {
+        std::string           reason0;
+        Worker::ActivityLevel workerLevel = Worker::computeActivityLevel(reason ? &reason0 : nullptr);
```

Comment:
Using a reference instead of a pointer will avoid ugly weird syntax like this...

Reply:
c.f. above reply. I am open to fix it (the feature) up by changing it to reference.